src/arch/linux/mapping/mapping.c

```diff
@@ -855,7 +853,10 @@ static int do_register_hwram(int type, unsigned base, unsigned size,
   hw->vbase = va;
   hw->size = size;
   hw->type = type;
-  hw->aliasmap = alloc_aliasmap(uaddr, size);
+  if (base + size > ALIAS_SIZE)
+    hw->aliasmap = alloc_aliasmap(uaddr, size);
+  else
+    hw->aliasmap = &lowmem_aliasmap[base >> PAGE_SHIFT];
```

Comment:
Because of the overlap with low 'v' mappings, i.e.:
`src/plugin/console/console.c:  register_hardware_ram('v', VGA_PHYS_TEXT_BASE, TEXT_SIZE(co,li));`
This works in devel, since the 'v' mapping prioritizes over the 'L' mapping and so you get the correct unix addr.

But if lowmem_aliasmap is used those maps need to be merged.

Reply:
no it's not the letter. It's how to handle the 'v' case. 

This would come in with base = 0xa0000, size = 0x20000 after and then just set `lowmem_aliasmap = hw->aliasmap;` (`if (base + size <= ALIAS_SIZE)`) for an incorrect map of 128k, and if use use `type == 'L'` instead, the 'v' aliasmap will be ignored by `dosaddr_to_unixaddr()`

Reply:
I think we are not on the same page. :)
Forget a letters for a moment.
Will this code work?
```
  if (base + size <= ALIAS_SIZE)
    lowmem_aliasmap = hw->aliasmap;
```
?
No letters here, as you see.

Reply:
since base + len = 0xa0000 + 0x20000 = 0xc0000 <= ALIAS_SIZE